### DISCUSSION TITLE
Did you test AveTemp with your CPU? Let me know

### DISCUSSION THREAD

**Discussion by treepoint** [treepoint/AveTemp#16] [Q&A]
I don't have many computers for testing so i don't know where AveTemp working well neither where it's not working at all, except for:

- Ryzen 4600H/4800H
- Intel Pentium 7505 Gold

I will appreciate if you would write there your CPU name and how AT is working with it.
——————

У меня недостаточно компьютеров для качественного тестирования, потому я не знаю где AveTemp работает хорошо, а где нет за исключением:

- Ryzen 4600H/4800H
- Intel Pentium 7505 Gold

Я был бы благодарен если бы вы написали название вашего процессора и как AT работает с ним.

**Comment by treepoint** [comment]
11th Gen Intel Core i7-1165G7 — doing well

**Comment by treepoint** [comment]
Hi, thanks for your feedback! I am not seeing your replies in discussion, only received its on mail. Can you tell me, please, did you delete them or there is some Github problems? 

What about Intel: i know about that bug, but can't reproduce it due to lack of intel machine with target CPU — Intel CPU load show incorrect on some CPU's · Issue #35 · treepoint/AveTemp (github.com). I can prebuild some special version of AveTemp, which dump some debug info to file and then, maybe, i can fix that problem if you will send me that file. 

What about newer version: i just released new version within morning, maybe it's that update. 

P.S. It's kinda strange when i am answering to nobody)

**Reply by guesswho2778** [reply]
sorry, i deleted them as i somehow downloaded the wrong version of your program. (i forgot about github emailing people about messages or else i would have just left them up and added to them)

i am however still seeing the that issue with the new version, im happy to send over some logs if you want them.
its getting pretty late where i am and it takes a solid 20 minutes to download, so sorry, but this ill do this in the morning.

**Reply by treepoint** [reply]
Thanks for reply! Morning or some delays — not a problem at all, cause i don't have enough time to start fixing things right in moment, that cause some delays from my side too. 

But i am happy to figure out intel problems with your help!

Btw i cooked debug version on AveTemp, you can download it here — https://drive.google.com/file/d/1qEnKlxLzqfidxi7o4syrRWPFHputuzZG/view?usp=share_link

I hope that windows link (AveTemp.exe - hardware dump) will work and all you need is:
1. Stop your copy of AveTemp.
2. Run that link, wait while AT start.
3. Hold on couple seconds, near exe should occur file «hardware_dump.txt» and start increasing size.
4. Stop that version AveTemp. If not stop, it will continue log to file, that unnecessary at our case.
5. Send me file «hardware_dump.txt».

If file will not occur or AT does not start, it means that link didn't work, and you need to create link manually. All you need is to start AT with parameter «--hardware-dump», just like on the picture and do the above sequence starts from 2 item.

![image](https://user-images.githubusercontent.com/25937222/226556160-5689cb7a-6ffa-4ea2-ada5-cdfb9e22b1c6.png)

**Reply by guesswho2778** [reply]
[hardware_dump.txt](https://github.com/treepoint/AveTemp/files/11026513/hardware_dump.txt)
there you go, there were quotation marks missing from the shortcut, so it didnt work right away.
based on what i read in that log it can see the usage, its just not outputting it to the ui on the application for some reason.
its also not showing even a blank space for thread usage, there are only four spots instead of eight
![image](https://user-images.githubusercontent.com/35134110/226560839-3dff134e-e92d-4a5d-a8ef-9c01bc881657.png)
hopefully this is helpful

ive also just noticed this in the notification tray.
![image](https://user-images.githubusercontent.com/35134110/226561246-0bb64519-5cf0-44a7-96f3-267abc9bf858.png)
having some options for that to select its colour along with other things like power usage, cpu usage and igpu usage might be a nice feature in the future.
for example, ive got this set up using msi afterburner, it allows you to see what is what at a glance.
![Uploading image.png…]()

**Reply by guesswho2778** [reply]
![image](https://user-images.githubusercontent.com/35134110/226562081-49caa6da-957d-4751-b602-bf7f859da9b8.png)
this is that last image that didnt upload properly (at least i dont see it on my end)

**Reply by treepoint** [reply]
Thanks! I briefly see log, and i think that i can figure it out, but can't say when will did it. Within week, i guess. As soon as new test version will be done, i'll text here.

> having some options for that to select its colour along with other things like power usage, cpu usage and igpu usage might be a nice feature in the future.

Agree, i made the task for this — https://github.com/treepoint/AveTemp/issues/54

**Reply by guesswho2778** [reply]
![Screenshot (2)](https://user-images.githubusercontent.com/35134110/226574406-3b061c02-a99d-49ef-a7b6-0100f09a6c47.png)

[hardware_dump.txt](https://github.com/treepoint/AveTemp/files/11027010/hardware_dump.txt)

got my old laptop up and running to test this on it and im seeing the same issue.
it has an i5-6200u

**Reply by treepoint** [reply]
Yeah, info log similar, except for count of core&threads. Only thing that i didn't understand for now — why only 2 load values in any problematic CPU. But i'll figure it out.

What about «there are only four spots instead of eight» and, as i seen above «two spots instead of four» it's feature, not a bug :D

The matter is in any SMT/HT process we're having doubled threads to cores, and LibHardwareMonitor get me info about 4 cores and 8 threads, for example, not about 8/8. So, i were forced to decide in which kind of representation we should implement it in UI: 

- full list of threads (and doubled cores), what may be a problem for 8 and 12 cores CPU's — 16 and 24 threads respectively. In this way we would need to show totally similar frequency for two chained threads.
- or with list of cores (and half threads) and calculate load for every core like average between 2 threads.

I decided to implement it like a list of cores — second option — cause its more universal for my opinion, and it have more useful info for square of UI space.

So, technically, we have problems with i5-6200u, but not with representation but with calculation.

**Reply by guesswho2778** [reply]
ah, i see.
that seems like the better, more compact and easier option to do.

**Reply by treepoint** [reply]
Hello!

Can you test this version and text me back is good or bad — https://drive.google.com/file/d/15gu5FZwVL9bKsbiBxrm4UjYBL9LA8xDV/view?usp=sharing ?

I think that Intel bug with load was fixed. If so, i will add that fix to next public version.

**Reply by guesswho2778** [reply]
![image](https://user-images.githubusercontent.com/35134110/227540044-f5063dbe-313f-4048-9be1-967ab84bc5b7.png)
that did indeed fix it, thanks.
